tools/functional-tester/etcd-runner/watcher.go

```diff
@@ -149,18 +115,56 @@ func performWatchOnPrefixes(ctx context.Context, getClient getClientFunc, round
 		rc.Close()
 	}
 
-	deletePrefixWithRety(client, ctx, roundPrefix)
+	deletePrefix(ctx, client, roundPrefix)
+}
+
+func checkWatchResponse(ctx context.Context, wc clientv3.WatchChan, prefix string, keys []string) {
+	for n := 0; n < len(keys); {
+		select {
+		case wr := <-wc:
+			for _, event := range wr.Events {
+				expectedKey := prefix + "-" + keys[n]
+				receivedKey := string(event.Kv.Key)
+				if expectedKey != receivedKey {
+					log.Fatalf("expected key %q, got %q for prefix : %q\n", expectedKey, receivedKey, prefix)
+				}
+				n++
+			}
+		case <-ctx.Done():
+			return
+		}
+	}
+}
+
+func putKey(ctx context.Context, client *clientv3.Client, key string) error {
```

Comment:
but it's guarded by the txn so it'll only bump the revision by at most 1-- maybe `putKeyAtMostOnce`?